src/Moryx.Runtime.Kernel/FileSystem/LocalFileSystem.cs

```diff
@@ -0,0 +1,278 @@
+﻿using Castle.MicroKernel.Registration;
+using Microsoft.Extensions.Logging;
+using Moryx.FileSystem;
+using Moryx.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Moryx.Runtime.Kernel.FileSystem
+{
+    internal class LocalFileSystem : IMoryxFileSystem
+    {
+        private string _fsDirectory;
+        private string _ownerFilesDirectory;
+        private readonly ILogger _logger;
+
+        private readonly List<MoryxFileTree> _ownerTrees = new List<MoryxFileTree>();
+
+        public LocalFileSystem(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger(nameof(LocalFileSystem));
+        }
+
+        public void SetBasePath(string basePath = "fs")
+        {
+            _fsDirectory = Path.Combine(Directory.GetCurrentDirectory(), basePath);
+            _ownerFilesDirectory = Path.Combine(_fsDirectory, "owners");
+        }
+
+        public void LoadTrees()
+        {
+            // Load all trees from the owner directory
+            var ownerFiles = Directory.EnumerateFiles(_ownerFilesDirectory);
+            foreach (var ownerFile in ownerFiles)
+            {
+                var treeHash = File.ReadAllText(ownerFile);
+                var ownerTree = new MoryxFileTree
+                {
+                    FileName = ownerFile,
+                    Hash = treeHash,
+                };
+                ReadExtensibleTree(ownerTree);
+            }
+        }
+
+        public MoryxFile GetFile(string hash)
+        {
+            return _ownerTrees.FindFile(hash);
+        }
+
+        public async Task<string> WriteAsync(MoryxFile file, Stream content)
+        {
+            if (file is MoryxFileTree fileTree)
```

Comment:
Shouldn't we throw ArgumentNullExceptions early, if e.g the file is null?

Reply:
Same goes for other method implementing the interface, `OpenStream` would run in a `NullRefException` for example🙈